File: liveDataDemo/app/src/main/java/edu/cs4730/livedatademo/DataViewModel.java
package edu.cs4730.livedatademo;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

/**
 * this is a simple viewmodel class holding two data items.
 */

public class DataViewModel extends ViewModel {

    MutableLiveData<Integer> count = new MutableLiveData<Integer>(0);
    MutableLiveData<String> name = new MutableLiveData<String>("");

    void setCount (int c) {count.setValue(c); }
    String getCount() {
        return String.valueOf(count.getValue());
    }

    void setName(String item) {
        name.setValue(item);
    }
    String getName() {
        return name.getValue();
    }


    public LiveData<Integer> getDataCount() {
        return count;
    }

    public LiveData<String> getDataName() {
        return name;
    }


    void incrementCount() {

        int c = count.getValue();
        c++;
        count.setValue(c);  //must use the setValue otherwise the observer will not be triggered!!!!
    }

}

File: liveDataDemo_kt/app/src/main/java/edu/esp/livedatademo_kt/DataViewModel.kt
package edu.esp.livedatademo_kt

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData

/**
 * this is a simple viewmodel class holding two data items.
 */
class DataViewModel : ViewModel() {
    var count = MutableLiveData(0)
    var name = MutableLiveData("")
    val dataCount: LiveData<Int>
        get() = count
    val dataName: LiveData<String>
        get() = name

    fun getCount(): String {
        return count.value.toString()
    }

    fun getName(): String? {
        return name.value
    }

    fun incrementCount() {
        var c = count.value!!
        c++
        count.value = c //must use the setValue otherwise the observer will not be triggered!!!!
    }

    fun setName(item: String) {
        name.value = item
    }
}